### DISCUSSION TITLE
Can I duplicate a plugin with a different name?

### DISCUSSION THREAD

**Discussion by amirhhashemi** [tailwindlabs/tailwindcss#8675] [Help] [answered]
The only downside of the typography plugin is that it doesn't support RTL so I have to change some of its classes in order to make it appropriate for RTL content. But what if I want to support RTL and LTR at the same time? Can I somehow keep the original plugin and add another variant of it with a different name? I mean I want the `prose` class to do whatever it already does, but I also want another variant of that, let's say `prose-rtl` that is based on the original typography plugin but it's optimized for RTL content. Then I can do something like this: `calss="rtl:prose-rtl ltr:prose"`. Is that possible? If yes, what's the cleanest way to do that?

**Comment by rohmann** [comment, marked as answer]
The way the typography plugin is setup is that you can pass in a custom `className` like this:

```
module.exports = {
  plugins: [
    require('@tailwindcss/typography')({
      className: 'prose-rtl'
    })
  ],
}
```

If you forked the plugin and made an RTL version, you could technically include them both and they'd be namespaced, provided that you gave them each a separate className.

I'm not sure if this is on the Tailwind roadmap but a great way to naturally support various writing modes in CSS is using logical box properties like [`padding-inline-start`](https://developer.mozilla.org/en-US/docs/Web/CSS/padding-inline-start). Instead of juggling multiple left/right padding values, you set `padding-inline-start: 20px` and it puts `20px` of padding at the beginning of the content based on the current text direction.

If you're planning on making a fork already, it might be worth exploring keeping it as one plugin and just updating the styles (https://github.com/tailwindlabs/tailwindcss-typography/blob/master/src/styles.js) to use those logical properties instead. That may give you more milage than having each mode in a separate plugin and would also result in a smaller CSS output.

**Reply by amirhhashemi** [reply]
Thanks. CSS logical properties seem to work for me. I'm wondering why they don't use it by default?

**Reply by rohmann** [reply]
🤷‍♂️ Not sure. Maybe it just hasn't been taken into consideration. I know it's only been about 2 years since all major browsers support it.
